Tests/SwiftParserTest/IsValidIdentifierTests.swift

```diff
@@ -0,0 +1,45 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftParser
+import XCTest
+
+class IsValidIdentifierTests: XCTestCase {
+  func testIsValidVariableName() {
+    XCTAssertTrue(isValidIdentifier("test", for: .variableName))
+    XCTAssertFalse(isValidIdentifier("class", for: .variableName))
+    XCTAssertTrue(isValidIdentifier("`class`", for: .variableName))
+    XCTAssertFalse(isValidIdentifier("self", for: .variableName))
+    XCTAssertTrue(isValidIdentifier("`self`", for: .variableName))
+    XCTAssertTrue(isValidIdentifier("`let`", for: .variableName))
+    XCTAssertFalse(isValidIdentifier("", for: .variableName))
+    XCTAssertFalse(isValidIdentifier("test: Int", for: .variableName))
+    XCTAssertFalse(isValidIdentifier("test ", for: .variableName))
+    XCTAssertFalse(isValidIdentifier("test\0", for: .variableName))
+    XCTAssertFalse(isValidIdentifier("test\0test", for: .variableName))
+  }
+
+  func testIsValidMemberAccessName() {
+    XCTAssertTrue(isValidIdentifier("test", for: .memberAccess))
+    XCTAssertTrue(isValidIdentifier("class", for: .memberAccess))
+    XCTAssertTrue(isValidIdentifier("`class`", for: .memberAccess))
+    XCTAssertFalse(isValidIdentifier("self", for: .memberAccess))
+    XCTAssertTrue(isValidIdentifier("`self`", for: .memberAccess))
+    XCTAssertTrue(isValidIdentifier("`let`", for: .memberAccess))
```

Comment:
`let` as well? Same above

Reply:
Ah, nice. If you're going that far you could have the init just take a map rather than each case, but it's just a test so I what you have is also fine 👍